samples/boards/nrf/s2ram/src/main.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+
+#include <zephyr/pm/pm.h>
+#include <zephyr/pm/policy.h>
+
+#include <zephyr/arch/common/pm_s2ram.h>
+
+#include <hal/nrf_power.h>
+#include <hal/nrf_regulators.h>
+#include <hal/nrf_gpio.h>
+
+static void system_off(void)
+{
+	nrf_regulators_system_off(NRF_REGULATORS);
+}
+
+static void do_suspend(void)
+{
+	irq_lock();
```

Comment:
This is not really necessary, `pm_state_set` is called with interruptions locked.